[dynamo, nested graph breaks] fix RETURN_VALUE tx skipping in nested graph breaks

williamwen42 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #163503
* #165597
* #163335
* #166016
* #165809
* __->__ #165808
* #166015
* #166013

Previously, we would completely skip building and calling any resume function if the leaf frame's resume instruction was RETURN_VALUE/RETURN_CONST. Now, we only skip building/calling resume functions for frames that are resuming on RETURN_VALUE.


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @Lucaskabela

williamwen42 committed:
[dynamo, nested graph breaks] fix RETURN_VALUE tx skipping in nested graph breaks

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, nested graph breaks] fix RETURN_VALUE tx skipping in nested graph breaks"


Previously, we would completely skip building and calling any resume function if the leaf frame's resume instruction was RETURN_VALUE/RETURN_CONST. Now, we only skip building/calling resume functions for frames that are resuming on RETURN_VALUE.


[ghstack-poisoned]

Lucaskabela reviewed (approved):
Thanks for the comments they are helpful in trying to parse out what all these stack operations and instruction pushes do.

As a minor nit, they seem a bit shorthanded still - could we transform them to long hand so new users can onboard to this logic easier?  i.e

```
do the same with cells
```
becomes
```
Filter out the cells of skipped tx'es from codegen
```

williamwen42 committed:
Update on "[dynamo, nested graph breaks] fix RETURN_VALUE tx skipping in nested graph breaks"


Previously, we would completely skip building and calling any resume function if the leaf frame's resume instruction was RETURN_VALUE/RETURN_CONST. Now, we only skip building/calling resume functions for frames that are resuming on RETURN_VALUE.


[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, nested graph breaks] fix RETURN_VALUE tx skipping in nested graph breaks"


Previously, we would completely skip building and calling any resume function if the leaf frame's resume instruction was RETURN_VALUE/RETURN_CONST. Now, we only skip building/calling resume functions for frames that are resuming on RETURN_VALUE.


[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, nested graph breaks] fix RETURN_VALUE tx skipping in nested graph breaks"


Previously, we would completely skip building and calling any resume function if the leaf frame's resume instruction was RETURN_VALUE/RETURN_CONST. Now, we only skip building/calling resume functions for frames that are resuming on RETURN_VALUE.


[ghstack-poisoned]
